### PR TITLE
Updating generator modification form

### PR DESCRIPTION
**Pull request by ayolab**
Signed-off-by: Ayoub LABIDI <ayoub.labidi@protonmail.com>

### CODE REVIEW

File: src/components/dialogs/generator-modification-dialog.js [hunk 1]
```diff
@@ -168,6 +197,66 @@ const GeneratorModificationDialog = ({
...
+        defaultValue:
+            (getValueOrNull(formValues?.reactiveCapabilityCurve) === null &&
+                generatorInfos?.minMaxReactiveLimits !== undefined) ||
+            getValueOrNull(formValues?.reactiveCapabilityCurve) === false
+                ? 'MINMAX'
+                : 'CURVE',
+        possibleValues: REACTIVE_LIMIT_TYPES,
+    });
+
+    const isReactiveCapabilityCurveOn = useCallback(() => {
```
[the remark this answers is not in the document]
Comment (ayolab) [reply src/components/dialogs/generator-modification-dialog.js]: you are right, changes done

File: src/components/dialogs/inputs/input-hooks.js [hunk 2]
```diff
@@ -436,7 +436,15 @@ export const useEnumValue = ({
                     ))}
                 </Select>
                 {previousValue && (
-                    <FormHelperText>{previousValue}</FormHelperText>
+                    <FormHelperText>
+                        {previousValue && doTranslation ? (
```
Comment (EstherDarkish) [inline src/components/dialogs/inputs/input-hooks.js:440]: The second previousValue test seems redondant

File: src/components/dialogs/inputs/reactive-capability-curve-table.js [hunk 3]
```diff
@@ -33,6 +33,44 @@ function hasValueCorrectFormat(valueToTest, index) {
     );
 }
 
+function getPreviousValue(displayedPreviousValues, displayedValues, index) {
```
Comment (EstherDarkish) [inline src/components/dialogs/inputs/reactive-capability-curve-table.js:36]: Can you explain the logic behind this function ? Why do we get the last array field in the first IF instead the indexed one ?

File: src/components/dialogs/inputs/reactive-capability-curve-table.js [hunk 4]
```diff
@@ -33,6 +33,44 @@ function hasValueCorrectFormat(valueToTest, index) {
...
+            p: displayedPreviousValues[index].p,
+            qminP: displayedPreviousValues[index].qminP,
+            qmaxP: displayedPreviousValues[index].qmaxP,
+        };
+    } else {
+        return null;
+    }
+}
+
+function fixValuesFormat(valuesToFix) {
```
Comment (EstherDarkish) [inline src/components/dialogs/inputs/reactive-capability-curve-table.js:64]: This function is bad. Why is it here ?
There is a bug if the value test is equal to zero. There are validation functions available for the job in the validation-functions.js file if needed.

**Review by EstherDarkish — CHANGES_REQUESTED** [review]

**Review by laurgarn — APPROVED** [review]

**Pull request merged**